### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Preadditive.ProjectiveResolution

### PR DESCRIPTION
**Pull request by joelriou**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Preadditive/ProjectiveResolution.lean [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+end ProjectiveResolution
+
+section
+
+variable [HasZeroMorphisms C] [HasZeroObject C] [HasEqualizers C] [HasImages C]
+
+def projectiveResolution (Z : C) [HasProjectiveResolution Z] : ProjectiveResolution Z :=
+  HasProjectiveResolution.out.some
+
+-- porting note: this was named `projectiveResolution` is mathlib. As there was also a need
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Preadditive/ProjectiveResolution.lean:318]:
```suggestion
-- porting note: this was named `projective_resolution` in mathlib 3. As there was also a need
```

File: Mathlib/CategoryTheory/Preadditive/ProjectiveResolution.lean [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+-- for a definition of `ProjectiveResolution Z` given `(Z : projectiveResolution Z)`, it
+-- seemed more consistent to have `projectiveResolution Z : ProjectiveResolution Z`
+-- and `projectiveResolution.complex Z : ChainComplex C ℕ`
+/-- An arbitrarily chosen projective resolution of an object. -/
+abbrev projectiveResolution.complex (Z : C) [HasProjectiveResolution Z] : ChainComplex C ℕ :=
+  (projectiveResolution Z).complex
+#align category_theory.projective_resolution CategoryTheory.projectiveResolution
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Preadditive/ProjectiveResolution.lean:325]: I think this means that the align should be like this, right?
```suggestion
#align category_theory.projective_resolution CategoryTheory.projectiveResolution.complex
```

Comment (joelriou) [reply Mathlib/CategoryTheory/Preadditive/ProjectiveResolution.lean]: Thanks very much for spotting this!

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**